### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by nathfreder**
This pull request upgrades all dependencies and allows this preset to work on the latest version of Neutrino. You will find that some small changes have also been made and they are listed in [`CHANGELOG.md`](https://github.com/soulchainer/neutrino-preset-styled-jsx/blob/24c1c4a68bb531a78be1386e5e29b719e99d3ef1/CHANGELOG.md).

### CODE REVIEW

**Comment by nathfreder** [comment]
@soulchainer

File: README.md [hunk 1]
```diff
@@ -1,58 +1,53 @@
...
-        'neutrino-preset-react',
-        'neutrino-preset-styled-jsx',
-        ...
-      ]
-    };
-  ```
-
-  And that's all. For `styled-jsx` specific usage, see their [docs][styled-url].
-  For Neutrino usage, refer to their [documentation][neutrino-docs].
-
```
Comment (soulchainer) [inline README.md]: Please, respect the badges being showed (don't remove badges already present, even without any explanation or warning) and also respect the way the links are constructed. Also would be good to respect the clarifying links.Thank you.

**Comment by nathfreder** [comment]
I am quite busy at the moment, thanks for your review and I hope to get back to work later on 😉.

**Comment by nathfreder** [comment]
@soulchainer I have completed your requests and added some things! Sorry about disregarding your preferences, I honestly did not know it would upset you. I will remember that next time. 😀

**Comment by soulchainer** [comment]
@nathfreder Totally forgot, man. I'm so sorry :(. Busy months, you know.
It's late night here and I'm sick, but I'm also on holidays :þ, so probably tomorrow I will check it properly (but everything seems fine to me right now) and merge/publish to npm and all that, if there isn't any prob I don't notice right now because I'm a little spaced out due to the cold.
Thanks for waiting for me.

↑ I wrote all that last night, but It seems I was so spaced out that I even forgot to hit «comment« ^_^U.
Sorry for the inconvenience. As said, I will try to get this sorted today. Have a nice day ;).

**Comment by nathfreder** [comment]
Thanks so much, 👍

**Comment by soulchainer** [comment]
@nathfreder Hi again.

Babel 7 was added here in a few dependencies, but after checking this and all that I found an error in web console, and then, checking the `styled-jsx` repository, I found this comment → https://github.com/zeit/styled-jsx/issues/458#issuecomment-401623620
So, basically looks like `styled-jsx` isn't compatible with Babel 7 yet, so update it the way it is currently would break things, I guess ^_^U.

I just made a easy check and run into this, so... >_<.

For checking the preset I:

- Checked to a branch with your code in the project. For that (GitHub itself tips me about this), from my project:
  `git checkout -b nathfreder-3.0.0 master`
  `git pull https://github.com/nathfreder/neutrino-preset-styled-jsx.git 3.0.0`
- Then I created a base Neutrino project with react, with:
   `npx @neutrinojs/create-project Test_Project`
- Install in the base neutrino project the preset from the local branch and also install `styled-jsx`:
  `cd Test_Project`
  `yarn add styled-jsx`
  `yarn add --dev file:/local/path/to/my/neutrino-preset-styled-jsx`
- Edit the `.neutrinorc.js` to use the preset (here you can see I choosed airbnb and jest in the `create-project` options):

  ```js
    module.exports = {
      use: [
        '@neutrinojs/airbnb',
        [
          '@neutrinojs/react',
          {
            html: {
              title: 'Test_Project'
            }
          }
        ],
        '@neutrinojs/jest',
        'neutrino-preset-styled-jsx'
      ]
      };
  ```

Then I added dummy code to check `styled-jsx`. I just edited `src/App.jsx`:

  ```js
    import { Component } from 'react';
    import './App.css';

    export default class App extends Component {
      state = {
        name: 'Test_Project',
      };

      render() {
        return (
          <div className="App">
            <h1>Welcome to {this.state.name}</h1>
            <style jsx>
              {`
                h1 { color: purple }
              `}
            </style>
          </div>
        );
      }
    }
  ```

Visually it works and it compiles, but in console you get some errors. And, like you can read in the issue I quoted you, probably this will throw other kind of errors in other situations.

TLDR: Seems for @giuseppeg reply (https://github.com/zeit/styled-jsx/issues/458#issuecomment-419386889) that there won't be support to Babel 7 in `styled-jsx` till v4 of the forementioned.
So, If you want this merged, I guess the faster way right now should be to stay in Babel 6.x, if possible, because Babel 7 could break things :).

Thanks for your patience and understanding ^^.

File: package.json [hunk 2]
```diff
@@ -24,36 +24,28 @@
...
   },
   "devDependencies": {
-    "babel-jest": "^21.0.0",
-    "babel-preset-env": "^1.6.1",
-    "commitizen": "^2.9.6",
-    "cz-conventional-changelog": "^2.0.0",
-    "jest": "^21.0.0",
-    "neutrino": "^7.0.0",
-    "webpack": "^3.0.0"
+    "@babel/core": "^7.1.2",
```
Comment (soulchainer) [inline package.json]: Babel 7 here and in a few other places. Which conflicts with `styled-jsx`, which doesn't support Babel 7, and causes unexpected behaviour.

**Comment by giuseppeg** [comment]
@soulchainer Babel 7 support can be added in a minor release too. It is just a matter of finding somebody who has time to make it happen :) If any of you folks have time I can explain what needs to be done

**Comment by nathfreder** [comment]
Alright, I'll bump it back to Babel `6.x.x`.

**Comment by nathfreder** [comment]
I *think* I fixed the issue, the tests are currently running.

**Comment by soulchainer** [comment]
Sorry for being so DAMN LATE with this, @nathfreder.
At the time, there was already passed some time since I ceased using Neutrino, so it let me forgot about this easily :sweat_smile:.

Anyways, I think, at the time, there were still remaining the issues I mentioned (even when the tests were passing, they weren't covering the exposed problem).

Anyways, like this is currently outdated, that «doesn't matter» really xD.
I will try to update this to the last version this weekend, if I can, and fix any other issue.

Then, maybe, edit the readme or whatever calling for maintainers or stuff, if necessary.

Sorry again for the long, long, long time T_T.
